charts/kubearchive/templates/api_server.yaml

```
@@ -22,6 +22,8 @@ spec:
           command: ["./go/bin/dlv"]
           args: ["--listen=:40000", "--headless=true", "--api-version=2", "--log", "exec", "/ko-app/api"]
           {{- end}}
+          env:
+{{ include "kubearchive.v1.otel.env" .Values.apiServer | indent 12 }}
```

I'm wondering if using the `template` directive will also work while keeping things simpler (I'm new to helm so I might not understand how this should be done, just read [this section](https://helm.sh/docs/chart_template_guide/named_templates/#declaring-and-using-templates-with-define-and-template) of the docs.

```suggestion
{{- template "kubearchive.v1.otel.env" .Values.apiServer }}
```

---

I was going through that page on how templates work. The last section is about [include](https://helm.sh/docs/chart_template_guide/named_templates/#the-include-function). If we use the `template` directive it renders the template directly into the yaml and it will be formatted according to the indentation in its definition. The docs recommend using include because it will render the template into a string that can be passed into other functions. This allows the users of the templates to format the template as needed for their use. My goal is to be able to reuse this template for other deployments (like the sink) that will need otel environment variables and may have a different level of indentation. I know it's kind of ugly but this is what is stated as best practice and allows for easier reuse

---

ok now I understand better the reason, thank you for the thorough explanation :)